app/controllers/api/v1/punches_controller.rb

```diff
@@ -1,6 +1,6 @@
 module Api
   module V1
-    class PunchesController < ApiController
+    class PunchesController < ActionController::API
```

Comment:
I changed the class used here as it was not possible to pass a custom authorization on the endpoint with the `ApiController` class. In order to maintain the custom authentication that `ApiController` implements, it is necessary to review the code to work with Swagger.